protocols/mdns/src/behaviour/iface.rs

```diff
@@ -198,43 +199,49 @@ impl InterfaceState {
         params: &impl PollParameters,
     ) -> Option<(PeerId, Multiaddr, Instant)> {
         // Poll receive socket.
-        while self.recv_socket.poll_readable(cx).is_ready() {
-            match self
-                .recv_socket
-                .recv_from(&mut self.recv_buffer)
-                .now_or_never()
-            {
-                Some(Ok((len, from))) => {
-                    if let Some(packet) = MdnsPacket::new_from_bytes(&self.recv_buffer[..len], from)
-                    {
-                        self.inject_mdns_packet(packet, params);
-                    }
+        match self
+            .recv_socket
+            .poll_receive_packet(cx, &mut self.recv_buffer)
```

Comment:
I think this should be a loop that reads for as long as it returns `Poll::Ready`.

Reply:
@thomaseizinger  the buffer is to long to require a loop?